go/private/mode.bzl

```diff
@@ -150,3 +150,61 @@ def mode_tags_equivalent(l, r):
             l.goarch == r.goarch and
             l.race == r.race and
             l.msan == r.msan)
+
+# Ported from https://github.com/golang/go/blob/master/src/cmd/go/internal/work/init.go#L76
+_LINK_C_ARCHIVE_PLATFORMS = {
+    "darwin/arm": None,
+    "darwin/arm64": None,
+}
+
+_LINK_C_ARCHIVE_GOOS = {
+    "dragonfly": None,
+    "freebsd": None,
+    "linux": None,
+    "netbsd": None,
+    "openbsd": None,
+    "solaris": None,
+}
+
+_LINK_C_SHARED_PLATFORMS = {
+    "linux/amd64": None,
+    "linux/arm": None,
+    "linux/arm64": None,
+    "linux/386": None,
+    "linux/ppc64le": None,
+    "linux/s390x": None,
+    "android/amd64": None,
+    "android/arm": None,
+    "android/arm64": None,
+    "android/386": None,
+}
+
+_LINK_PLUGIN_PLATFORMS = {
+    "linux/amd64": None,
+    "linux/arm": None,
+    "linux/arm64": None,
+    "linux/386": None,
+    "linux/s390x": None,
+    "linux/ppc64le": None,
+    "android/amd64": None,
+    "android/arm": None,
+    "android/arm64": None,
+    "android/386": None,
+    "darwin/amd64": None,
+}
+
+def link_mode_args(mode):
+    platform = mode.goos + "/" + mode.goarch
+    args = []
+    if mode.link == LINKMODE_C_ARCHIVE:
+        if platform in _LINK_C_ARCHIVE_PLATFORMS or mode.goos in _LINK_C_ARCHIVE_GOOS:
+            if platform == "linux/ppc64":
```

Comment:
This `return` could cause problems. Can the condition be folded into the `if` above? `... and platform != "linux/ppc64"`

Reply:
Sure!